webapp/channels/src/components/widgets/settings/text_setting.tsx

```diff
@@ -73,6 +73,21 @@ export default class TextSetting extends React.PureComponent<WidgetTextSettingPr
                     disabled={this.props.disabled}
                 />
             );
+        } else if (type === undefined) {
+            input = (
+                <input
+                    autoFocus={this.props.autoFocus}
+                    data-testid={this.props.id + type}
+                    id={this.props.id}
+                    className='form-control'
+                    type='input'
+                    placeholder={this.props.placeholder}
+                    value={this.props.value}
+                    maxLength={this.props.maxLength}
+                    onChange={this.handleChange}
+                    disabled={this.props.disabled}
+                />
+            );
```

Comment:
so why did we add this here?

Reply:
in the else block the line `type = ['input', 'email', 'tel', 'number', 'url', 'password'].includes(type) ? type : 'input';` will show error if undefined is passed in includes function that is why I handled undefined case separately

Reply:
if thats the case then lets pass correct `type` in TextSetting of `webapp/channels/src/components/admin_console/elasticsearch_settings.tsx`